### PR TITLE
fix issue #26

### LINKED ISSUE #26: Optimize Drop Course Logic
- Problem: when dropping a course, an error may occur due to invalid course ID or code, and it's inconvenient for user to manually configure all the course once again.
- One possible solution: Auto-detect and prompt the user to confirm whether a course is dropped or not.

### PR DESCRIPTION
**Pull request by jackye666**
Fix #26. 
Requirements satisfied:
If you drop a course , then next time you use this project, it will automatically delete the courseID and courseCode from the config file and inform the user.
Design idea:
If the course is dropped, then you cannnot successfully get the relative cousre information from canvas. 
For courses downloaded through courseID, I check the return json object. If the json object does not contain "id", then it means that we failed to access the course page. I then delete this courseID from the config.
For courses downloaded through courseCode, I create a list which keeps appending successfully downloaded coursesCode. Then, I compare the list with the courseCode in config file , and delete them.
Quality Assurance:
I conduct peer review with my teamate spikedyf and there is no obviou bug found. What's more, I test on the following test: 
I am in course "EECS 477 001 FA 2021" and not in "EECS 496". I enter both course code to the config, and it turns out that "EECS 496" is successfully deleted and the files from "EECS 477 001 FA 2021" are successfully downloaded.

### CODE REVIEW

File: canvassyncer/__main__.py [hunk 1]
```diff
@@ -187,6 +187,7 @@ async def getCourseIdByCourseCodeHelper(self, page, lowerCourseCodes):
         if not courses:
             return res
         for course in courses:
+            #print(course)
```
Comment (BoYanZh) [inline canvassyncer/__main__.py:192]: Commented debug lines should be deleted.

File: canvassyncer/__main__.py [hunk 2]
```diff
@@ -208,10 +209,41 @@ async def getCourseIdByCourseCode(self):
...
+                json.dump(data,f)
+                f.truncate()
     async def getCourseCodeByCourseIDHelper(self, courseID):
         url = f"{self.baseurl}/courses/{courseID}"
         sessRes = await self.sessGetJson(url)
+        #print("this is sessRes:\n ",sessRes)
+        if "id" not in sessRes.keys():
+            print("There is problem with courseID",courseID)
+            print("Deleted from config!")
+            with open(".canvassyncer.json","r+") as f:
```
Comment (BoYanZh) [inline canvassyncer/__main__.py]: Users may specify the config file through -p.

**Comment by BoYanZh** [comment]
Thanks for your contribution and observation. But I think it may be better to leave the config file unchanged but just prompt users that some course Code/course ID is unreachable, then skip these courses in the following steps. If you want to override the config files, it may break the config file under some special conditions.

**Comment by jackye666** [comment]
Thank you for your comments, I changed my code so that it only inform the user and won't change the config.

File: canvassyncer/__main__.py [hunk 3]
```diff
@@ -11,6 +11,8 @@
 import aiofiles
 import aiohttp
 from tqdm import tqdm
+global droppedCourse
+droppedCourse={"courseIDs":[],"courseCodes":[]}
```
Comment (BoYanZh) [inline canvassyncer/__main__.py:15]: Consider storing the value inside the class.

File: canvassyncer/__main__.py [hunk 4]
```diff
@@ -208,10 +211,23 @@ async def getCourseIdByCourseCode(self):
                     endOfPage = True
                 self.courseCode.update(item)
             page += PAGES_PER_TIME
+            correct_courseCode = []
```
Comment (BoYanZh) [inline canvassyncer/__main__.py:214]: Unify the naming style with the project. Use camel case.

File: canvassyncer/__main__.py [hunk 5]
```diff
@@ -208,10 +211,23 @@ async def getCourseIdByCourseCode(self):
...
+            for i in self.courseCode:
+                correct_courseCode.append(self.courseCode[i])
+            for i in self.config["courseCodes"]:
+                if i not in correct_courseCode:
+                    if i not in droppedCourse["courseCodes"]:
```
Comment (BoYanZh) [inline canvassyncer/__main__.py:219]: Combine consecutive `if` with `and`.

**Review by BoYanZh — CHANGES_REQUESTED** [review]

**Pull request closed without merging**